### ISSUE TITLE
Helm installation to AKS 1.19.7 fails

### ISSUE THREAD

**Issue by anttipo** [jenkinsci/kubernetes-operator#565]
**Describe the bug**
We are deploying the Jenkins operator to an AKS cluster running 1.19.7 via Helm v3. During the deployment, we can see the pods and resources being created just fine. However, once the helm release is over, we can see the deployment as failed:

```
helm history jenkins-operator -n jenkins
REVISION	UPDATED                 	STATUS    	CHART                 	APP VERSION	DESCRIPTION
1       	Thu May 13 08:17:30 2021	failed	jenkins-operator-0.4.3	0.5.0      	Release "jenkins-operator" failed: timed out waiting for the condition
```

This seems to happen without an apparent reason. We tried experimenting back and forth with some values, such as namespace and jenkins.enabled to no avail. Also increasing the timeout to 15min made no difference. CRDs are installed first separately and they succeed without a problem

However, after I rerun `helm upgrade --install` the deployment instantly succeeds:

```
helm history jenkins-operator -n jenkins
REVISION	UPDATED                 	STATUS    	CHART                 	APP VERSION	DESCRIPTION
1       	Thu May 13 08:17:30 2021	superseded	jenkins-operator-0.4.3	0.5.0      	Release "jenkins-operator" failed: timed out waiting for the condition
2       	Thu May 13 08:24:10 2021	deployed  	jenkins-operator-0.4.3	0.5.0      	Upgrade complete
```

**To Reproduce**
Simply try deploying the Jenkins operator to an AKS cluster 1.19.7 with Helm. We have been unable to evaluate the setup with a different k8s version due to some restrictions on our end.

**Additional information**

Kubernetes version: 1.19.7
Jenkins Operator version: Helm chart version 0.4.3

Add error logs about the problem here (operator logs and Kubernetes events).

This is the log output from the operator pod which doesnt change between the first failed and the second succeeded attempt at deploying via Helm:

```
2021-05-13T08:17:35.620Z	INFO	controller-jenkins.cmd	manager/main.go:55	Version: v0.5.0
2021-05-13T08:17:35.620Z	INFO	controller-jenkins.cmd	manager/main.go:56	Git commit: fe81e5a
2021-05-13T08:17:35.620Z	INFO	controller-jenkins.cmd	manager/main.go:57	Go Version: go1.14.2
2021-05-13T08:17:35.620Z	INFO	controller-jenkins.cmd	manager/main.go:58	Go OS/Arch: linux/amd64
2021-05-13T08:17:35.620Z	INFO	controller-jenkins.cmd	manager/main.go:59	operator-sdk Version: v0.17.0
2021-05-13T08:17:35.620Z	INFO	controller-jenkins.cmd	manager/main.go:85	Watch namespace: jenkins
2021-05-13T08:17:35.620Z	INFO	leader	leader/leader.go:46	Trying to become the leader.
2021-05-13T08:17:37.318Z	INFO	leader	leader/leader.go:88	No pre-existing lock was found.
2021-05-13T08:17:37.325Z	INFO	leader	leader/leader.go:108	Became the leader.
2021-05-13T08:17:38.029Z	INFO	controller-runtime.metrics	metrics/listener.go:40	metrics server is starting to listen	{"addr": "0.0.0.0:8383"}
2021-05-13T08:17:38.029Z	INFO	controller-jenkins.cmd	manager/main.go:110	Registering Components.
2021-05-13T08:17:40.851Z	WARN	controller-jenkins.cmd	manager/main.go:155	Could not generate and serve custom resource metrics	{"error": "discovering resource information failed for Route in route.openshift.io/v1: apiVersion route.openshift.io/v1 and kind Route not found available in Kubernetes cluster"}
2021-05-13T08:17:41.584Z	INFO	metrics	metrics/metrics.go:97	Metrics Service object created	{"Service.Name": "jenkins-operator-metrics", "Service.Namespace": "jenkins"}
2021-05-13T08:17:42.286Z	WARN	controller-jenkins.cmd	manager/main.go:174	Could not create ServiceMonitor object	{"error": "no ServiceMonitor registered with the API"}
2021-05-13T08:17:42.286Z	WARN	controller-jenkins.cmd	manager/main.go:178	Install prometheus-operator in your cluster to create ServiceMonitor objects	{"error": "no ServiceMonitor registered with the API"}
2021-05-13T08:17:42.286Z	INFO	controller-jenkins.cmd	manager/main.go:182	Starting the Cmd.
2021-05-13T08:17:42.286Z	INFO	controller-runtime.manager	manager/internal.go:356	starting metrics server	{"path": "/metrics"}
2021-05-13T08:17:42.286Z	INFO	controller-runtime.controller	controller/controller.go:164	Starting EventSource	{"controller": "jenkins-controller", "source": "kind source: jenkins.io/v1alpha2, Kind=Jenkins"}
2021-05-13T08:17:42.286Z	INFO	controller-runtime.controller	controller/controller.go:164	Starting EventSource	{"controller": "jenkinsimage-controller", "source": "kind source: /, Kind="}
2021-05-13T08:17:42.486Z	INFO	controller-runtime.controller	controller/controller.go:164	Starting EventSource	{"controller": "jenkinsimage-controller", "source": "kind source: /, Kind="}
2021-05-13T08:17:42.587Z	INFO	controller-runtime.controller	controller/controller.go:164	Starting EventSource	{"controller": "jenkins-controller", "source": "kind source: core/v1, Kind=Pod"}
2021-05-13T08:17:42.587Z	INFO	controller-runtime.controller	controller/controller.go:164	Starting EventSource	{"controller": "jenkins-controller", "source": "kind source: core/v1, Kind=Secret"}
2021-05-13T08:17:42.587Z	INFO	controller-runtime.controller	controller/controller.go:164	Starting EventSource	{"controller": "jenkinsimage-controller", "source": "kind source: /, Kind="}
2021-05-13T08:17:42.687Z	INFO	controller-runtime.controller	controller/controller.go:171	Starting Controller	{"controller": "jenkinsimage-controller"}
2021-05-13T08:17:42.687Z	INFO	controller-runtime.controller	controller/controller.go:164	Starting EventSource	{"controller": "jenkins-controller", "source": "kind source: core/v1, Kind=Secret"}
2021-05-13T08:17:42.688Z	INFO	controller-runtime.controller	controller/controller.go:190	Starting workers	{"controller": "jenkinsimage-controller", "worker count": 1}
2021-05-13T08:17:42.688Z	INFO	controller-runtime.controller	controller/controller.go:164	Starting EventSource	{"controller": "jenkins-controller", "source": "kind source: core/v1, Kind=ConfigMap"}
2021-05-13T08:17:42.688Z	INFO	controller-runtime.controller	controller/controller.go:171	Starting Controller	{"controller": "jenkins-controller"}
2021-05-13T08:17:42.688Z	INFO	controller-runtime.controller	controller/controller.go:190	Starting workers	{"controller": "jenkins-controller", "worker count": 1}
```

**Comment by prryb** [comment]
Hello, could you post Kubertenes events (`kubectl get events`)? That might help us find the root cause of the problem.

**Comment by Sig00rd** [comment]
We've tried to install the latest released (Helm Chart v0.5.0 with Operator v0.6.0) version of Operator on Kubernetes 1.19.7 cluster running on Minikube, using ```helm upgrade --install``` and we weren't able to reproduce this issue (the release was deployed successfully). One thing to note, however, is that CRDs are already in the Helm Chart and installing them separately might cause problems.

**Comment by anttipo** [comment]
Hi,

Sorry for the late response. We had initial issues when I posted the original issue but have been unable to reproduce it even myself later on. We also installed the CRDs as part of the Helm chart but since everything seems to be working fine now, I'll close the issue.

Thanks for your time!
